src/blockdb/wrapper.cpp

```diff
@@ -214,10 +237,26 @@ void SyncStorage(const CChainParams &chainparams)
     {
         std::vector<std::pair<int, CDiskBlockIndex> > indexByHeight;
         pblocktreeother->GetSortedHashIndex(indexByHeight);
+        LOGA("indexByHeight size = %u \n", indexByHeight.size());
         int64_t bestHeight = 0;
+        uint64_t lastFinishedFile = 0;
         CBlockIndex *pindexBest = new CBlockIndex();
+        // Load block file info
+        int loadedblockfile = 0;
+        pblocktreeother->ReadLastBlockFile(loadedblockfile);
+        LOGA("loadedblockfile = %i \n", loadedblockfile);
+        std::vector<CBlockFileInfo> blockfiles;
+        blockfiles.resize(loadedblockfile + 1);
+        LOGA("blockfiles.size() = %u \n", blockfiles.size());
+        for (int nFile = 0; nFile <= loadedblockfile; nFile++)
+        {
+            pblocktreeother->ReadBlockFileInfo(nFile, blockfiles[nFile]);
+        }
+
         for (const std::pair<int, CDiskBlockIndex> &item : indexByHeight)
         {
+            bool needData = true;
+            bool needUndo = true;
             CBlockIndex *index;
```

Comment:
i don't see that this flag ever gets modified anywhere,  it's always true from what I can see...do you really need it for something?